internal/db/advisory_locks.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2025 Canonical.
+
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+// LockDI represents a unique identifier for an advisory lock in the database.
+type LockID int64
+
+const (
+	// ControllerBootstrapLock is the advisory lock ID used for controller bootstrap operations.
+	ControllerBootstrapLock LockID = 1001
```

Comment:
You could use an iota here, something like the 
```suggestion
	ControllerBootstrapLock LockID = iota + 1
```

Reply:
Sure, added.